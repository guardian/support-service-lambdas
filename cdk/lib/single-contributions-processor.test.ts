import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {
	APP_NAME,
	SingleContributionsProcessor,
} from './single-contributions-processor';

describe('The SingleContributionsProcessor stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const codeStack = new SingleContributionsProcessor(
			app,
			`${APP_NAME}-CODE`,
			{ stack: 'membership', stage: 'CODE' },
		);
		const prodStack = new SingleContributionsProcessor(
			app,
			`${APP_NAME}-PROD`,
			{ stack: 'membership', stage: 'PROD' },
		);
		expect(Template.fromStack(codeStack).toJSON()).toMatchSnapshot();
		expect(Template.fromStack(prodStack).toJSON()).toMatchSnapshot();
	});
});
