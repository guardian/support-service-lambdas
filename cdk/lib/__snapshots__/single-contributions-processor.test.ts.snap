// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SingleContributionsProcessorStack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "singlecontributionsprocessordlqurl": {
      "Value": {
        "Ref": "singlecontributionsprocessorqueueCF05834A",
      },
    },
    "singlecontributionsprocessorqueueurl": {
      "Value": {
        "Ref": "deadletterssinglecontributionsprocessorqueueC118704C",
      },
    },
  },
  "Resources": {
    "deadletterssinglecontributionsprocessorqueueC118704C": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "dead-letters-single-contributions-processor-queue-CODE",
        "SqsManagedSseEnabled": true,
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "singlecontributionsprocessorqueueCF05834A": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "single-contributions-processor-queue-CODE",
        "RedrivePolicy": {
          "deadLetterTargetArn": {
            "Fn::GetAtt": [
              "deadletterssinglecontributionsprocessorqueueC118704C",
              "Arn",
            ],
          },
          "maxReceiveCount": 2,
        },
        "SqsManagedSseEnabled": true,
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;

exports[`The SingleContributionsProcessorStack matches the snapshot 2`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "singlecontributionsprocessordlqurl": {
      "Value": {
        "Ref": "singlecontributionsprocessorqueueCF05834A",
      },
    },
    "singlecontributionsprocessorqueueurl": {
      "Value": {
        "Ref": "deadletterssinglecontributionsprocessorqueueC118704C",
      },
    },
  },
  "Resources": {
    "deadletterssinglecontributionsprocessorqueueC118704C": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "dead-letters-single-contributions-processor-queue-PROD",
        "SqsManagedSseEnabled": true,
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "singlecontributionsprocessorqueueCF05834A": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "QueueName": "single-contributions-processor-queue-PROD",
        "RedrivePolicy": {
          "deadLetterTargetArn": {
            "Fn::GetAtt": [
              "deadletterssinglecontributionsprocessorqueueC118704C",
              "Arn",
            ],
          },
          "maxReceiveCount": 2,
        },
        "SqsManagedSseEnabled": true,
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/support-service-lambdas",
          },
          {
            "Key": "Stack",
            "Value": "membership",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
